backend/src/schema/resolvers/users.ts

```diff
@@ -150,6 +150,31 @@ export default {
         }
         params.termsAndConditionsAgreedAt = new Date().toISOString()
       }
+
+      const {
+        emailNotificationSettings,
+      }: { emailNotificationSettings: { name: string; value: boolean }[] | undefined } = params
+      delete params.emailNotificationSettings
+      if (emailNotificationSettings) {
+        emailNotificationSettings.forEach((setting) => {
+          const allowedSettingNames = [
```

Comment:
Why not defining this in the GraphQL Schema?

Reply:
How would you define such a komplex logic in the gql?

Reply:
I refer to the allowed names. Something like this:
```
enum NotificationReason {
  mentioned_in_post
  mentioned_in_comment
  commented_on_post
  user_joined_group
  user_left_group
  changed_group_member_role
  removed_user_from_group
}
```

Reply:
This would reduce the code of validation at this point